### PR TITLE
New cleanup plugin that deletes offset files after tests (closes #669)

### LINKED ISSUE #669: XDR offsets are not deleted after the tests
The tests that use XTC or TR files produce cached offsets for these trajectories (files names .*_offset.npz). These files remain in the `testsuite/MDAnalysisTests/data` directory and never get clened. In addition to being messy, it makes some tests fail after a git checkout (probably because it messes with the ctimes).

Here is an example of error:

```
======================================================================
ERROR: test_persistent_offsets_size_mismatch (MDAnalysisTests.coordinates.test_xdr.TestXTCReader_offsets)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/jon/dev/mdanalysis/testsuite/MDAnalysisTests/coordinates/test_xdr.py", line 765, in test_persistent_offsets_size_mismatch
    assert_equal(warn[0].message.args,
IndexError: list index out of range

----------------------------------------------------------------------
```

ping @kain88-de

### PR DESCRIPTION
**Pull request by mnmelo**

### CODE REVIEW

**Comment by kain88-de** [comment]
When is this plugin run? Does nose pick it up magically?

**Comment by mnmelo** [comment]
Sorry, I realize some documentation would have helped!

As it is now you must pass the `--with-mda_cleanup` flag to `mda_nosetests`.
I realize I also didn't add the flag to the travis builds.

I don't have any strong feelings for it being on or off by default, or whether to run it in travis (where it is unneeded since state is not kept between builds). What say you?

**Comment by kain88-de** [comment]
I think @jbarnoud wants it on by default.

**Comment by jbarnoud** [comment]
I do not really mind if it is activated by default or not. On one hand, having it by default will avoid running into problems when the offset format changes, on the other hand, we may not notice these problems.

I know I will activate it in my own testing script at least.

File: testsuite/MDAnalysisTests/plugins/cleanup.py [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+
+    def configure(self, options, conf):
+        super(Cleanup, self).configure(options, conf)
+        self.config = conf # This will let other tests know about config settings.
+
+    def finalize(self, result):
+        from .. import __name__ as mdatestsname
+        dirname = resource_filename(mdatestsname, 'data')
+        #print("Cleanup: looking for offsets to clean in {}".format(dirname))
+        for root, dirs, fnames in walk(dirname):
```
Comment (kain88-de) [inline testsuite/MDAnalysisTests/plugins/cleanup.py]: does this walk down directories?

Comment (jbarnoud) [inline testsuite/MDAnalysisTests/plugins/cleanup.py]: Yes it does.

**Comment by mnmelo** [comment]
Ok, I made it on by default and cleaned up other minor aspects.
It'll report what it's doing if nose's `verbosity` is greater than 2 (`-v` sets `verbosity=2`, so you'll need to specifically ask for `--verbosity=3` if you want to see anything.)

I also cleaned up minor things in the `error_capture` plugin. It might be a good idea to create our own subclass of Plugin that leaves a plugin on by default (right now `error_capture` and `cleanup` have some code duplication in this respect.)

And yes, it traverses subdirs. Here's the output on a testrun with `--verbosity=3`:
```
----------------------------------------------------------------------
Ran 3326 tests in 55.887s

OK (KNOWNFAIL=3, SKIP=5)
Cleanup: looking for offsets to clean in /home/manel/src/MDAnalysis/mdanalysis/testsuite/build/lib.linux-x86_64-2.7/MDAnalysisTests/data
Cleanup: deleting offset file /home/manel/src/MDAnalysis/mdanalysis/testsuite/build/lib.linux-x86_64-2.7/MDAnalysisTests/data/.adk_oplsaa.trr_offsets.npz
Cleanup: deleting offset file /home/manel/src/MDAnalysis/mdanalysis/testsuite/build/lib.linux-x86_64-2.7/MDAnalysisTests/data/.gram_A_identical_frames.xtc_offsets.npz
Cleanup: deleting offset file /home/manel/src/MDAnalysis/mdanalysis/testsuite/build/lib.linux-x86_64-2.7/MDAnalysisTests/data/.cobrotoxin.trr_offsets.npz
Cleanup: deleting offset file /home/manel/src/MDAnalysis/mdanalysis/testsuite/build/lib.linux-x86_64-2.7/MDAnalysisTests/data/.cobrotoxin.xtc_offsets.npz
Cleanup: deleting offset file /home/manel/src/MDAnalysis/mdanalysis/testsuite/build/lib.linux-x86_64-2.7/MDAnalysisTests/data/.adk_oplsaa.xtc_offsets.npz
Cleanup: deleting offset file /home/manel/src/MDAnalysis/mdanalysis/testsuite/build/lib.linux-x86_64-2.7/MDAnalysisTests/data/coordinates/.test.trr_offsets.npz
Cleanup: deleting offset file /home/manel/src/MDAnalysis/mdanalysis/testsuite/build/lib.linux-x86_64-2.7/MDAnalysisTests/data/coordinates/.test.xtc_offsets.npz
```

**Comment by kain88-de** [comment]
Can you change the verbosity to be enabled at level 1. Ideally level 0. We are doing something non-standard here and I think we should inform devs who use this later early no to avoid surprises for them.

**Comment by mnmelo** [comment]
I can do it, but are you sure you want this to show up even when we're at low verbosity? I think it kills the simplicity of it and of having the summary as last output line. But I don't really care about it.
(I can also try making this show up before the summary).

This is how the non-verbose default would look like:
```
....S...................S....................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................K...............................K...................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................S..SS....................................................................................................................................................................................................................................................K...................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................
----------------------------------------------------------------------
Ran 3326 tests in 56.123s

OK (KNOWNFAIL=3, SKIP=5)
Cleanup: looking for offsets to clean in /home/manel/src/MDAnalysis/mdanalysis/testsuite/build/lib.linux-x86_64-2.7/MDAnalysisTests/data
Cleanup: deleting offset file /home/manel/src/MDAnalysis/mdanalysis/testsuite/build/lib.linux-x86_64-2.7/MDAnalysisTests/data/.adk_oplsaa.trr_offsets.npz
Cleanup: deleting offset file /home/manel/src/MDAnalysis/mdanalysis/testsuite/build/lib.linux-x86_64-2.7/MDAnalysisTests/data/.gram_A_identical_frames.xtc_offsets.npz
Cleanup: deleting offset file /home/manel/src/MDAnalysis/mdanalysis/testsuite/build/lib.linux-x86_64-2.7/MDAnalysisTests/data/.cobrotoxin.trr_offsets.npz
Cleanup: deleting offset file /home/manel/src/MDAnalysis/mdanalysis/testsuite/build/lib.linux-x86_64-2.7/MDAnalysisTests/data/.cobrotoxin.xtc_offsets.npz
Cleanup: deleting offset file /home/manel/src/MDAnalysis/mdanalysis/testsuite/build/lib.linux-x86_64-2.7/MDAnalysisTests/data/.adk_oplsaa.xtc_offsets.npz
Cleanup: deleting offset file /home/manel/src/MDAnalysis/mdanalysis/testsuite/build/lib.linux-x86_64-2.7/MDAnalysisTests/data/coordinates/.test.trr_offsets.npz
Cleanup: deleting offset file /home/manel/src/MDAnalysis/mdanalysis/testsuite/build/lib.linux-x86_64-2.7/MDAnalysisTests/data/coordinates/.test.xtc_offsets.npz
```

**Comment by kain88-de** [comment]
I would activate it with simple verbosity.  That should be enough

**Comment by mnmelo** [comment]
Ok, that sounds like a good compromise (and will make it show up in travis). Still, I'll try to move output before the summary line.

**Comment by mnmelo** [comment]
ping @kain88-de Is this ok for merging? It's blocking the 0.14 release...

**Comment by mnmelo** [comment]
Hmm, I just noticed that travis' verbosity changed from what I expected. I'm quickly updating the verbosity limits.

**Comment by kain88-de** [comment]
Looks  good. Thanks for fixing it.

**Pull request merged**